// axioos/src/page/Login.js
import "../App.css"
import { Form, Input, Button} from 'antd';
// import React,{useEffect,useState} from 'react'
import axios from "axios"
import {useHistory} from "react-router-dom"
export default function Login() {
    const history = useHistory()
    const onFinish = async (values) => {
        let res = await axios.get("/api/list",{params:{values:values}});
        if(res.data.code === 200){
            alert("登录成功")
            localStorage.setItem("user",values.username);
            history.push("/app")
        }else{
            alert("登录失败")
        }
        console.log(values);
        // setValues(values)
    };

    // const onFinishFailed = (errorInfo) => {
    //     console.log('Failed:', errorInfo);
    //     alert("密码或账号输入错误")
    // };    
//    let [value,setValues] = useState("")

    // useEffect(  async ()=>{
    //     let res = await axios.get("/api/list",{params:{value}});
    //     console.log(res.data);
    // },[])
   
    return (
        <div className="login">
            <div className="con">
                <Form
                    layout="horizontal"
                    name="basic"
                    labelCol={{
                        span: 5,
                    }}
                    wrapperCol={{
                        span: 16,
                    }}
                    initialValues={{
                        remember: true,
                    }}
                    onFinish={onFinish}
                    // onFinishFailed={onFinishFailed}
                >
                    <Form.Item
                        label="用户名"
                        name="username"
                        rules={[
                            {
                                required: true,
                                message: '请输入用户名!',
                            },
                        ]}
                    >
                        <Input />
                    </Form.Item>

                    <Form.Item
                        label="密码"
                        name="password"
                        rules={[
                            {
                                required: true,
                                message: '请输入密码!',
                            },
                        ]}
                    >
                        <Input.Password />
                    </Form.Item>
                    <Form.Item
                        wrapperCol={{
                            offset: 8,
                            span: 20,
                        }}
                    >
                        <Button type="primary" htmlType="submit">
                            点击登录
                        </Button>
                    </Form.Item>
                </Form>
            </div>
        </div>
    )
}
